bgpd/bgp_routemap.c

```diff
@@ -802,26 +802,50 @@ struct route_map_rule_cmd route_match_mac_address_cmd = {
 	"mac address", route_match_mac_address, route_match_mac_address_compile,
 	route_match_mac_address_free};
 
-/* `match vni' */
-
-/* Match function should return 1 if match is success else return
-   zero. */
+/*
+ * Match function returns:
+ * ...RMAP_MATCH if match is found.
+ * ...RMAP_NOMATCH if match is not found.
+ * ...RMAP_OKAY to bypass this check..
+ */
 static route_map_result_t route_match_vni(void *rule,
 					  const struct prefix *prefix,
 					  route_map_object_t type, void *object)
 {
 	vni_t vni = 0;
+	unsigned int label_cnt = 0;
 	struct bgp_path_info *path = NULL;
+	struct prefix_evpn *evp = (struct prefix_evpn *) prefix;
 
 	if (type == RMAP_BGP) {
 		vni = *((vni_t *)rule);
 		path = (struct bgp_path_info *)object;
 
+		/*
+		 * This rmap filter is valid for vxlan tunnel type only.
+		 * For any other tunnel type, return RMAP_OKAY, so, we don't
+		 * RMAP_DENY this, which can happen by returning
+		 * RMAP_MATCH or RMAP_NOMATCH.
+		 */
+		if (path->attr && path->attr->encap_tunneltype !=
+			BGP_ENCAP_TYPE_VXLAN)
+			return RMAP_OKAY;
+
+		/*
+		We do not want to filter type 3 routes because
```

Comment:
we have 2 different comment styles in the same function!  Please use the same comment style as on like 824+

Reply:
Agreed.

Reply:
Agreed